src/main/java/org/ngafid/common/TimeUtils.java

```diff
@@ -114,13 +117,30 @@ public static String toUTC(String date, String time, String offset) {
 
     }
 
+    /**
+     * Helper method to parse the date and time using the declared formatters
+     * @param dateTimeString
+     * @return
+     */
+    public static LocalDateTime parseDateTime(String dateTimeString) {
+        String normalizedDateTime = dateTimeString.replaceAll("\\s+", " ");  // Replaces multiple spaces with a single space
+        for (DateTimeFormatter formatter : DATE_FORMATTERS) {
+            try {
+                return LocalDateTime.parse(normalizedDateTime, formatter);
+            } catch (Exception e) {
+                // Continue trying other formats
+            }
+        }
+        // If none of the formats work, throw an exception or handle it appropriately
+        throw new IllegalArgumentException("Date format not supported: " + normalizedDateTime);
```

Comment:
Please stop using this exception type. Here a FlightFileFormat exception would work, or you could create a new checked exception type.